### ISSUE TITLE
FLAC__bitwriter_write_byte_block "could be faster"

### ISSUE THREAD

**Issue by jherby2k** [xiph/flac#86]
I'm profiling a performance bottleneck in my .NET Core application (calls libFLAC via p/invoke), and it looks like the FLAC encoder is spending a good 70% of its time in this method, after I call `FLAC__stream_encoder_init_stream`. This happens when I write a (pretty large, ~3MB) picture block along with the audio.

I expect some overhead, but noticed the comment in the code:

```c
inline FLAC__bool FLAC__bitwriter_write_byte_block(FLAC__BitWriter *bw, const FLAC__byte vals[], uint32_t nvals)
{
	uint32_t i;

	/* this could be faster but currently we don't need it to be since it's only used for writing metadata */
	for(i = 0; i < nvals; i++) {
		if(!FLAC__bitwriter_write_raw_uint32_nocheck(bw, (FLAC__uint32)(vals[i]), 8))
			return false;
	}

	return true;
}
```

I'm not a C guy... anyone able to take a stab at improving this method?

**Comment by enzo1982** [comment]
@erikd, @jherby2k: One of you might want to close this now that 1.3.3 is released with the #94 performance fix.

**Comment by jherby2k** [comment]
Fixed by #94
